### PR TITLE
postgres_plugin: database size chart added, code refactoring

### PR DESCRIPTION
**Pull request by l2isbad**
https://github.com/firehol/netdata/issues/1750
it was a quick fix.:
1. index and table stats charts are disabled by default 
2. database size (per database) added

Per database metrics are not wrong anymore... But there is a lot of room for improvements.

### CODE REVIEW

**Comment by ktsaou** [comment]
merge?

**Comment by facetoe** [comment]
So you just disabled the metrics? Most of these changes seem unnecessary for the goal...

**Comment by l2isbad** [comment]
> Perhaps they should be disabled by default if an alternative, global source of information can't be found?

Of course. The guy said "per database metrics are not wrong" and you confirmed it. Why we need them on by default?
And he suggested to add 

> update the plugin to display size of database (reuse the same connection that today used plus the pg_stat_database view and pg_database_size function). This information can be displayed per database or for the whole cluster.

Added.

As i say postgres plugin can be improved in many ways. I have not enough knowledge of postgres right now to do it. I will look at it later.

Now it's just a fix - remove misleading charts. All other changes to make the code simpler from my point of view.

**Comment by facetoe** [comment]
I think we can retain the metrics by using the `pg_table_size` and `pg_index_size` views. I need to run some tests to determine how expensive these queries will be but if they are within a suitable range I think this would be preferable to removing the metrics completely. 

@ktsaou are you able to hold off on merging these changes until after the weekend? I'd like to have the chance to investigate retaining these metrics using the above views. If the queries are not too expensive I will submit a PR.

**Comment by l2isbad** [comment]
To be honest i don't like `format, get` wall and different methods where the logic always the same.
My logic was
1. **Create one** query: metric dict
2. **Create one** function to get data from all queries.

That is all. Works ok.

I think we can merge changes and you can apply whatever you want after.

Are you ok with that @facetoe ?

File: python.d/postgres.chart.py [hunk 1]
```diff
@@ -217,91 +267,53 @@ def _connect(self):
         return True
 
     def check(self):
+        if not PSYCOPG2:
+            self.error('\'python-psycopg2\' module is needed to use postgres.chart.py')
```
Comment (facetoe) [inline python.d/postgres.chart.py]: This is good, much more informative message for the user.

File: python.d/postgres.chart.py [hunk 2]
```diff
@@ -217,91 +267,53 @@ def _connect(self):
...
+            return False
+        if not self._connect():
+            self.error('Can\'t connect to %s' % str(self.configuration))
```
Comment (facetoe) [inline python.d/postgres.chart.py]: My concern with this is that it could leak database user/password information in the logs.

Comment (l2isbad) [inline python.d/postgres.chart.py]: Agree.

**Comment by facetoe** [comment]
This might be a personal preference @l2isbad but I don't feel that many of these changes contribute to making the module easier to read or understand. Now I'm no doubt biased as I wrote the original code so maybe someone else can offer their opinion on whether or not the PR increases the readability/efficiency/maintainability of the module?

File: python.d/postgres.chart.py (hunk 2 above)
Comment (facetoe) [inline python.d/postgres.chart.py]: Looks like the original also had this defect. Might be an idea to just sanitize the configuration before printing as it's still useful information for debugging.

**Comment by l2isbad** [comment]
Yeah. It's always personal. That is why i rewrite it. 
1. All metrics at the top - much simpler to understand what is collected.
2. Since all metrics defined before collection i can use one simple 'for' loop to get them all (functions with same logic and different names in your case)

super simple

**Comment by l2isbad** [comment]
@ktsaou 
i suggest to merge

**Comment by ktsaou** [comment]
merged

**Pull request merged**